jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/Callback.java

```diff
@@ -437,45 +435,7 @@ static Callback combine(Callback cb1, Callback cb2)
         if (cb2 == null)
             return cb1;
 
-        return new Callback()
-        {
-            @Override
-            public void succeeded()
-            {
-                try
-                {
-                    cb1.succeeded();
-                }
-                finally
-                {
-                    cb2.succeeded();
-                }
-            }
-
-            @Override
-            public void failed(Throwable x)
-            {
-                try
-                {
-                    cb1.failed(x);
-                }
-                catch (Throwable t)
-                {
-                    if (x != t)
-                        x.addSuppressed(t);
-                }
-                finally
-                {
-                    cb2.failed(x);
-                }
-            }
-
-            @Override
-            public InvocationType getInvocationType()
-            {
-                return Invocable.combine(Invocable.getInvocationType(cb1), Invocable.getInvocationType(cb2));
-            }
-        };
+        return from(cb1, cb2);
```

Comment:
The `from(Callback cb1, Callback cb2)` implementation is more brittle than the one it replaces here as it does not try to manage the exceptions.

IMHO the `from(Callback cb1, Callback cb2)` implementation should become what has been deleted here.

Reply:
I'm not sure there is a correct behaviour when callbacks start throwing... but yeah, I switched it around